src/utils/Subscription.js

```diff
@@ -1,45 +1,61 @@
 // encapsulates the subscription logic for connecting a component to the redux store, as
 // well as nesting subscriptions of descendant components, so that we can ensure the
 // ancestor components re-render before descendants
```

Comment:
Yeah. My thought was to wait until these changes have landed, take care of any outstanding issues and let the dust settle before further tinkering.